sdk/src/main/java/io/radar/sdk/util/RadarLogBuffer.kt

```diff
@@ -0,0 +1,199 @@
+package io.radar.sdk.util
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import io.radar.sdk.Radar
+import io.radar.sdk.model.RadarLog
+import org.json.JSONObject
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.util.*
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.ReentrantLock
+
+/**
+ * Wraps the filesystem to support log caching
+ */
+internal class RadarLogBuffer(
+    private val context: Context,
+    private val executor: ExecutorService = Executors.newSingleThreadExecutor()
+) {
+
+    private companion object {
+        const val DIRECTORY = "RadarLogs"
+
+        /**
+         * Maximum number of logs in the buffer.
+         */
+        const val LOG_BUFFER_SIZE_LIMIT = 500
+    }
+
+    /**
+     * Lock for handling changes to [currentFile]
+     */
+    private val fileLock = ReentrantLock()
+
+    /**
+     * Lock for handling changes to the files within [DIRECTORY]
+     */
+    private val directoryLock = ReentrantLock()
+
+    /**
+     * Contains the name of the current log file
+     */
+    private var currentFile: String
+
+    /**
+     * Number of log entries in the current log file.
+     */
+    val size: AtomicInteger
+
+    init {
+        //First find current file
+        val files = getDirectory().listFiles()
+        if (files?.isNotEmpty() == true) {
+            //This gets the latest file that was in use if it can, otherwise it returns the first one.
+            currentFile = files.maxByOrNull { getLastModified(it) }!!.name
+            //This is initialized to the number of lines in the current file
+            size = AtomicInteger(FileInputStream(File(getDirectory(), currentFile)).bufferedReader().readLines().size)
```

Comment:
Might just be the API interfaces, but do we really need a `bufferedReader` if we're loading all the lines into memory with `readLines` anyways?

Reply:
Would this be more succinct?

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#lines-java.nio.file.Path-

Reply:
@jkao looks like that method is not supported at the level of Android we support. It does seem excessive just to get the number of lines in the file, however this only occurs when the log buffer object is initialized, and the files should not be very lengthy.